static/js/settings_account.js

```diff
@@ -362,17 +373,27 @@ export function set_up() {
             $(this).attr("href", settings_bots.encode_zuliprc_as_uri(data));
         });
 
-        $("#api_key_modal [data-dismiss=modal]").on("click", () => {
+        $("#api_key_modal [data-micromodal-close]").on("click", () => {
             common.reset_password_toggle_icons(
                 "#get_api_key_password",
                 "#get_api_key_password + .password_visibility_toggle",
             );
         });
-    });
+    };
 
     $("#api_key_button").on("click", (e) => {
+        $("body").append(render_settings_api_key_modal());
         setup_api_key_modal();
-        overlays.open_modal("#api_key_modal");
+        $("#api_key_status").hide();
+        overlays.open_modal("api_key_modal", {
+            autoremove: true,
+            micromodal: true,
+            micromodal_opts: {
+                onShow: () => {
+                    $("#get_api_key_password").trigger("focus");
+                },
+            },
+        });
```

Comment:
What's the thinking behind not using `dialog_widget` for this one?  Not sure I have a clear preference but we've been converting almost all of these to those.

Reply:
We need different HTML for the modal buttons than present in `dialog_widget`.
It could be done using `dialog_widget` but it would be messy so I preferred not using it for this one.

Reply:
I guess this widget is pretty weird, and I'm hoping we'll get to delete it in any case when we move to supporting multiple API keys, so probably fine to do it like this given that it works :).